### PR TITLE
Upgrade to Angular 16

### PR DESCRIPTION
**Pull request by bjorn-einar-bjartnes-4ss**
Trying to upgrade to ng16, I have one issue remaining I might need some help with struggling to build the scss files (edit: I did some changes that know seem to work after my latest change, but they should be carefully reviewed as I don't know why I had to change the imports.)

image

Mostly been upgrading fairly standard, but struggling a bit as I haven't used yarn and instead of npm and I also don't know how this styles.scss file is supposed to be imported. It worked fine before I started the upgrade an ran tests, so it has something to do with the upgrade.

```
ng update @angular-eslint/eslint-plugin@15 @angular-eslint/eslint-plugin-template@15 @angular-eslint/schematics@15 @angular-eslint/template-parser@15

ng update @angular-eslint/eslint-plugin@16 @angular-eslint/eslint-plugin-template@16 @angular-eslint/schematics@16 @angular-eslint/template-parser@16
```

### CODE REVIEW

File: src/test.ts [hunk 1]
```diff
@@ -7,13 +7,7 @@ import {
   platformBrowserDynamicTesting,
 } from '@angular/platform-browser-dynamic/testing';
 
-declare const require: any;
-
 // First, initialize the Angular testing environment.
 getTestBed().initTestEnvironment(BrowserDynamicTestingModule, platformBrowserDynamicTesting(), {
     teardown: { destroyAfterEach: false }
 });
-// Then we find all the tests.
```
Comment (bjorn-einar-bjartnes-4ss) [inline src/test.ts:16]: This is something the migration scripts did, not me...

**Comment by bjorn-einar-bjartnes-4ss** [comment]
Have been trying all sorts of ```    "styleIncludePaths": ["../../node_modules"]``` and variations of these, but I honestly don't understand how this is meant to be configured.

**Comment by bjorn-einar-bjartnes-4ss** [comment]
> Have been trying all sorts of ` "styleIncludePaths": ["../../node_modules"]` and variations of these, but I honestly don't understand how this is meant to be configured.

I removed some tildes, just randomly trying stuff, and now it does seem to build and run all tests. I have no idea _why_, but it does build and run without them, so something has perhaps changed...?

**Comment by cmckni3** [comment]
> > Have been trying all sorts of ` "styleIncludePaths": ["../../node_modules"]` and variations of these, but I honestly don't understand how this is meant to be configured.
> 
> I removed some tildes in [70b82d9](https://github.com/webcat12345/ngx-ui-switch/commit/70b82d9093c12b53385f7735ad98287fd863b551), just randomly trying stuff, and now it does seem to build and run all tests. I have no idea _why_, but it does build and run without them, so something has perhaps changed...? Or, perhaps I broke all the styles. It looks fine when I run `ng serve`

Could be an ng-compiler change. Going to review this later.

**Comment by cmckni3** [comment]
Ugh, going to have to switch over the GitHub actions since Travis CI open source shut down in December 2020.

**Comment by bjorn-einar-bjartnes-4ss** [comment]
> Ugh, going to have to switch to GitHub actions since Travis CI open source shut down in December 2020.

(Not very much related, it just felt very much like this discussion from Twitter on the rot of cloud platforms. Ah, well. Let me know if there is anything I can help with...
https://twitter.com/Aaronontheweb/status/1682363458283601924 )

**Comment by cmckni3** [comment]
> > Ugh, going to have to switch to GitHub actions since Travis CI open source shut down in December 2020.
> 
> (Not very much related, it just felt very much like this discussion from Twitter on the rot of cloud platforms. Ah, well. Let me know if there is anything I can help with... https://twitter.com/Aaronontheweb/status/1682363458283601924 )

😂 

[This is exactly what happened](https://twitter.com/Aaronontheweb/status/1682381447632502784?s=20). Travis moved my cheese by shutting down the free service.

> Ah, well. Let me know if there is anything I can help with

Going to run the travis yaml through the GitHub actions converter. Should have it up and running in a decent amount of time.

**Comment by cmckni3** [comment]
Coming back to this. Will have an update soon

**Comment by cmckni3** [comment]
> > Have been trying all sorts of ` "styleIncludePaths": ["../../node_modules"]` and variations of these, but I honestly don't understand how this is meant to be configured.
> 
> I removed some tildes in [70b82d9](https://github.com/webcat12345/ngx-ui-switch/commit/70b82d9093c12b53385f7735ad98287fd863b551), just randomly trying stuff, and now it does seem to build and run all tests. I have no idea _why_, but it does build and run without them, so something has perhaps changed...? Or, perhaps I broke all the styles. It looks fine when I run `ng serve`

yeah this was the fix. The `~` meaning "the node_modules directory" was deprecated in ng 15.

https://stackoverflow.com/questions/73853771/scss-usage-of-in-imports-is-deprecated-angular-vscode

**Comment by cmckni3** [comment]
Resolved conflicts. I upgraded to 15.x in the meantime for a 15.x version

**Comment by cmckni3** [comment]
Running a `yarn install` to update lock file

**Review by cmckni3 — APPROVED** [review]

**Comment by cmckni3** [comment]
Thanks!

Saved me time having to debug the scss issue in stackblitz and locally

**Pull request merged**